ledger/src/entry/repo.rs

```diff
@@ -102,4 +100,44 @@ impl Entries {
 
         Ok(ret)
     }
+
+    pub async fn list_by_transaction_ids(
+        &self,
+        tx_ids: Vec<TransactionId>,
+    ) -> Result<HashMap<TransactionId, Vec<Entry>>, SqlxLedgerError> {
+        let tx_ids: Vec<Uuid> = tx_ids.into_iter().map(|id| Uuid::from(id)).collect();
+        let records = sqlx::query!(
+            r#"SELECT id, version, transaction_id, account_id, journal_id, entry_type, layer as "layer: Layer", units, currency, direction as "direction: DebitOrCredit", sequence, description, created_at, modified_at
+            FROM sqlx_ledger_entries
+            WHERE transaction_id = ANY($1)"#,
```

Comment:
You should add an `ORDER BY` clause to ensure the entries are in the correct order per transaction.